src/main/java/com/minecolonies/coremod/colony/permissions/Permissions.java

```diff
@@ -571,8 +573,7 @@ public void savePermissions(@NotNull final CompoundNBT compound)
     @Override
     public boolean hasPermission(final Rank rank, @NotNull final Action action)
     {
-        return (rank == ranks.get(OWNER_RANK_ID) && action != Action.GUARDS_ATTACK)
-                 || Utils.testFlag(permissionMap.get(rank), action.getFlag())
+        return Utils.testFlag(permissionMap.get(rank), action.getFlag())
```

Comment:
... while this means that owners don't automatically get su-style "ignore permission settings" for everything.  Which should mostly be fine, but they could lose permissions if they're set weird currently.  (Notably it also means that the owners who currently deliberately perform "lethal selection" on their colonists will suddenly not be able to any more until they change this setting...)

This is not necessarily a bad thing, and is probably intended.  But it seems worth pointing out.

Reply:
That probably applies to the above, serverside enforcement. To not allow the server owner from locking themselves out.

Reply:
No, by "lethal selection" I meant players intentionally killing citizens/visitors because they wanted different ones.  Just pointing out that this change will mean that owners will no longer be able to do that unless they explicitly enable the permission.

The other part is that prior to this change, the code would just grant all permissions (except `GUARDS_ATTACK`) regardless of whatever settings were in the town hall, and after this change the settings in the town hall will actually matter to owners.  This will be fine if the town hall settings are left at defaults, but if they've toggled things around to a silly configuration then it may affect them poorly.

As I said, these are both probably intended changes, but are worth noting and may surprise people and generate questions.

What _might_ be still worthwhile actually changing though is that while `togglePermission` protects against disabling the permissions that owners should always have, if an existing colony has some of these required owner permissions already missing then they're going to now take effect and the owner won't be able to fix their colony permissions manually.  (Unless there's already somewhere outside of this PR that fixes these permissions if missing; I haven't checked the rest of the code.)

Reply:
"No, by "lethal selection" I meant players intentionally killing citizens/visitors because they wanted different ones. Just pointing out that this change will mean that owners will no longer be able to do that unless they explicitly enable the permission."

Yes, gotta be either lava, or alter the permission. That's okay.

"The other part is that prior to this change, the code would just grant all permissions (except GUARDS_ATTACK) regardless of whatever settings were in the town hall, and after this change the settings in the town hall will actually matter to owners. This will be fine if the town hall settings are left at defaults, but if they've toggled things around to a silly configuration then it may affect them poorly."

Yes, that's why I tried to make sure they can't change any setting that could lock them out.

"What might be still worthwhile actually changing though is that while togglePermission protects against disabling the permissions that owners should always have, if an existing colony has some of these required owner permissions already missing then they're going to now take effect and the owner won't be able to fix their colony permissions manually. (Unless there's already somewhere outside of this PR that fixes these permissions if missing; I haven't checked the rest of the code.)"

Someone has to test this, from my tests, for the owner, everything defaults to on. But might be worth a test.

Reply:
> Someone has to test this, from my tests, for the owner, everything defaults to on. But might be worth a test.

For new colonies, sure.  But if you make a colony in an older version, can you toggle off those permissions there (which has no effect in that version) and then upgrade to this version and then oops, you're locked out?

Or even if the old version UI wouldn't let you do that either, if there's some corrupt data in the colony permissions and those are absent then nobody would have noticed before (since it would have worked anyway) but now their colony is unusable after upgrade.

Reply:
Old UI didn't have owner toggles.